user_funcs.py

```diff
@@ -116,3 +117,246 @@ def login(conn) -> tuple[str, int]:
+                case 1:
+                    curs.execute(remove_movie_query, (movie_ID, collection_ID_list[index][0]))
+
+                    # action for adding a movie
+                    # This does work
+                case 2:
+                    curs.execute(add_movie_query, (movie_ID, collection_ID_list[index][0]))
+
+                    # action to rename a collection
+                    # Fucntionality works
+                case 3:
+                    new_name = input("what would you like to name your collection: ")
+                    curs.execute(change_name_query, (new_name, collection_ID_list[index],))
+
+                    # action to delete a collection
+                case 4:
+                    #This does work
+
+                    curs.execute(delete_all_movie_query, (collection_ID_list[index],))
+                    curs.execute(delete_moviecollection_query, (collection_ID_list[index],))
+                case 5:
+                    return
+    conn.commit()
+    return
+
+def browse_collections(conn, user_ID) -> None:
+    """
+    A different version to print all information of a users collection/
+    list of collections
+
+    :param conn: Connection to the database
+    :param user_ID: The ID of the user
+    :return: Nothing
+    """
+     # A Query to get the collection ID for a certain User ID
+    query = """
+    SELECT collectionid FROM moviecollection as mc 
+    WHERE mc.madeby = %s 
+    ORDER BY name ASC
+    """
+    # A Query to get the name of the collection to be displayed
+    get_name_query = """
+    SELECT name FROM moviecollection
+    WHERE collectionid = %s
+    """
+    # A query to copunt the number of movies in a collection
+    count_movie_query = """
+    SELECT COUNT(movieid) FROM incollection
+    WHERE collectionid = %s
+    """
+    # A query to get all lengths of all movies in a collection
+    get_length_movie_query = """
+    SELECT SUM(length) FROM movie as m
+    WHERE m.movieid in (
+    SELECT movieid FROM incollection
+    WHERE collectionid = %s
+    )
+    """
+    print(f"Printing all collections belong to the user...")
+    collection_ID = ""
+    with conn.cursor() as curs:
+        curs.execute(query, (user_ID,))
+        #Get list of collection IDs for a user and iterate through them
+        collection_ID_list = curs.fetchall()
+        print(collection_ID_list)
+        for i in range (0, len(collection_ID_list)):
+
+            #Get name of current collection ID
+            collection_ID = int(collection_ID_list[i][0])
+            curs.execute(get_name_query, (collection_ID,))
+            collection_name = curs.fetchone()[0]
+
+            #Get number of movies for current collection ID
+            curs.execute(count_movie_query, (collection_ID,))
+            num_movie = curs.fetchone()[0]
+
+            #Get total length of all movies in the collection (in minutes)
+            curs.execute(get_length_movie_query, (collection_ID,))
```

Comment:
If you use the above suggestions to combine queries, you shouldn't need to do any queries in a for loop (this is generally considered bad practice and I'm pretty sure prof johnson is gonna execute us if he sees this).